### ISSUE TITLE
Build that passed earlier today fails after being rerun

### ISSUE THREAD

**Issue by jksaunders** [travis-ci/travis-ci#9761]
My project started having unexpected failed tests in my builds (passing locally) about an hour ago. Out of curiosity, I reran one of my passing builds (7 hours ago) and it failed in the same tests my builds are currently failing.

It's a Grails 3 project, failing during tests that were passing 7 hours ago on the same commit. I've tried clearing the project cache but still seeing the issue. Any ideas at all?

Config if it's helpful:
```
{
  "os": "linux",
  "jdk": "oraclejdk8",
  "dist": "trusty",
  "cache": {
    "directories": [
      "$HOME/.gradle",
      "node_modules"
    ]
  },
  "group": "stable",
  "script": "./travis.sh",
  "language": "groovy",
  "before_install": [
    "sudo apt-get update -qq",
    "chmod +x ./gradlew",
    "chmod +x ./grailsw",
    "chmod +x ./travis.sh"
  ]
}
```

**Comment by jksaunders** [comment]
Seems to have resolved itself 🤷‍♂️

**Comment by karussell** [comment]
We probably have the same problem. But we can't figure out what the underlying problem is and it persists. It has no proper error message and occurs only for oraclejdk8: https://github.com/graphhopper/graphhopper/issues/1397

**Comment by jksaunders** [comment]
After some digging, it turned out to be application code that was doing some poor assumptions about dates 🤦‍♂️ So then the actual issue was that 

1. I had an edge case in some application code re: date/time
2. Locally, the the date/time was not the edge case
3. On the Travis CI instance, the date/time did fall into the edge case

So that explained why I couldn't get it to reproduce locally, and also explains why old passing builds failed on rerun (at the time of the original run, the date/time wasn't in the edge case, and when I reran it, the date/time did fall into the edge case.

TL;DR it was my bad, but the root cause of my not being able to reproduce it locally ended up being a time difference between local environment and the Travis CI environment. Perhaps there's a similar difference between your environment and your Travis CI build environments.

I checked out your Travis setup and commented on your cited issue for Graphhopper!
